model.py

```
@@ -207,5 +207,79 @@ def forward(self, x, time_embedding):
+        input_attn = input_attn.transpose(1, 2) # ensure that channels is at end to apply attention
+
+        output_attn, _ = self.attention_block_multihead(input_attn, input_attn, input_attn)
+        output_attn = output_attn.transpose(1, 2).reshape(B, C, H, W)   # reshape after briging back the colors dim to match input
+        
+        out += output_attn  # add back the residual connection
+        return out
+        ################################# Same as DownBlock ################################# 
+    
+        
+class UNET(nn.Module):
+    def __init__(self, in_channels, down_channels=[32, 64, 128, 256], bottleneck_channels=[256, 256, 128], embedding_dim=128, down_sample=[True, True, True]):
```

Ruff (0.8.0)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)